client/me/purchases/remove-plan-dialog/index.tsx

```diff
@@ -0,0 +1,182 @@
+import { Dialog, Gridicon } from '@automattic/components';
+import { useTranslate } from 'i18n-calypso';
+import { Fragment } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import { SiteExcerptData } from 'calypso/data/sites/site-excerpt-types';
+import { isRefundable } from 'calypso/lib/purchases';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import { SiteScreenshot } from '../site-screenshot/';
+import getPlanFeatures from './get-plan-features';
+import './style.scss';
+
+interface RemovePlanDialogProps {
+	closeDialog: () => void;
+	removePlan: () => void;
+	isDialogVisible: boolean;
+	isRemoving: boolean;
+	site: SiteExcerptData;
+	hasDomain: boolean;
+	primaryDomain: string;
+	wpcomURL: string;
+}
+
+export const RemovePlanDialog = ( {
+	closeDialog,
+	removePlan,
+	isDialogVisible,
+	site,
+	hasDomain,
+	primaryDomain,
+	wpcomURL,
+}: RemovePlanDialogProps ) => {
+	const translate = useTranslate();
+	const siteName = site.name ?? '';
+
+	/**
+	 * Istantiate site's plan variables.
+	 */
+	const productSlug = site.plan?.product_slug;
+	const planLabel = site.plan?.product_name_short;
+	const isComingSoon = site.is_coming_soon;
+	const isPrivate = site.is_private;
+	const launchedStatus = site.launch_status === 'launched' ? true : false;
+	const shouldUseSiteThumbnail =
+		isComingSoon === false && isPrivate === false && launchedStatus === true;
+	const subTitle = ! isRefundable
```

Comment:
`isRefundable` is a function, but you're using it like a variable -- that may be why this code isn't working correctly?

Reply:
@JanaMW27 ok, this could be a good compromise

Reply:
Similar to what @JanaMW27 said below, I wouldn't recommend truncating their custom domain.

If you want to truncate the `.wordpress.com` domain, I would at least do it at a significantly longer word length?  I think it can be a fair amount longer than you have it now without looking terrible.  And at that point, we'd also be talking about a very tiny number of sites, so I'm not sure how much it really matters... (For what it's worth, you could also just use `overflow-wrap` CSS to keep it from looking like a complete disaster and then not worry about the rest.)

Reply:
All right, I'm doing it